### PR TITLE
Module builder

### PR DESCRIPTION
**Pull request by pjfanning**
https://github.com/FasterXML/jackson-module-scala/issues/525

### CODE REVIEW

File: src/main/scala/com/fasterxml/jackson/module/scala/deser/TupleDeserializerModule.scala [hunk 1]
```diff
@@ -5,7 +5,7 @@ import com.fasterxml.jackson.databind._
 import com.fasterxml.jackson.databind.deser.std.StdDeserializer
 import com.fasterxml.jackson.databind.deser.{BeanDeserializerFactory, Deserializers}
 import com.fasterxml.jackson.databind.jsontype.TypeDeserializer
-import com.fasterxml.jackson.module.scala.JacksonModule
```
Comment (cowtowncoder) [inline src/main/scala/com/fasterxml/jackson/module/scala/deser/TupleDeserializerModule.scala:8]: Just noticed this -- not wrong per se, but I wonder if there might be confusion now that Jackson 3.0 has renamed `Module` as `[com.fasterxml.jackson.databind.]JacksonModule`?

Comment (pjfanning) [reply src/main/scala/com/fasterxml/jackson/module/scala/deser/TupleDeserializerModule.scala:8]: It is a different class that does extend com.fasterxml.jackson.databind.JacksonModule (or Module). I'm open to renaming the Scala JacksonModule class but I'm not sure it is necessary.

**Comment by cowtowncoder** [comment]
Nothing much to comment since I am not familiar enough to offer much useful feedback.
Just one question: from the way it is passed, the "builder" sounds more like it might be a Configuration/settings object?
If so, maybe it'd be make sense to name it as such? (if I am mistaken about intent, apologies).

File: src/main/scala/com/fasterxml/jackson/module/scala/deser/TupleDeserializerModule.scala (hunk 1 above)
Comment (cowtowncoder) [reply src/main/scala/com/fasterxml/jackson/module/scala/deser/TupleDeserializerModule.scala:8]: Right, understood. It is more an internal implementation detail it seems.
So naming is definitely up to you, I just thought I'd mention it.

**Comment by pjfanning** [comment]
@cowtowncoder this 'Builder' class follows the Builder pattern and is used to build a ScalaModule (which extends the JacksonModule/Module)

**Comment by cowtowncoder** [comment]
@pjfanning Ok then I am confused. From changes it seemed odd that it would be passed to (de)serializers and value instantiators and so on. I'll see if I can grok the flow with more context.

**Comment by pjfanning** [comment]
I can rename the ReadOnlyBuilder trait as Config - would make that part of the code more readable

**Comment by cowtowncoder** [comment]
@pjfanning on builder arg passed, yes, I can see now how it adds things to be added via SetupContext.

**Comment by cowtowncoder** [comment]
@pjfanning Ah. On `ReadOnlyBuilder` as base -- not sure what makes most sense. Given that it really is an internal base type I don't have a strong opinion here.

**Pull request merged**